src/StakeSablierNFT.sol

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.8.19;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Adminable } from "@sablier/v2-core/src/abstracts/Adminable.sol";
+import { ISablierV2Lockup } from "@sablier/v2-core/src/interfaces/ISablierV2Lockup.sol";
+
+/// @title StakeSablierNFT
+/// @notice DISCLAIMER: This template has not been audited and is provided "as is" with no warranties of any kind,
+/// either express or implied. It is intended solely for demonstration purposes on how to build a staking contract using
+/// Sablier NFT. This template should not be used in a production environment. It makes specific assumptions that may
+/// not be applicable to your particular needs.
+/// @dev This template allows users to stake Sablier NFTs and earn staking rewards.
+///
+/// Requirements:
+///   - The Sablier NFT must be transferrable.
+///
+/// Risks:
+///   - The sender MUST NOT call `withdraw()` for staked NFTs. If the sender calls `withdraw()` for staked NFTs, the
+///     funds will be sent to the staking contract and will be locked forever.
+///   - The sender MUST NOT cancel the stream for staked NFTs. If the sender cancels the stream for staked NFTs, the
```

Comment:
Because I wanted to demonstrate both the following:
- Allow staking when `isCancelable` is true.
- Allow staking when `isCancelable` is false.

We can discourage but we cannot stop them to allow staking of cancelable streams.